extensions/cornerstone/src/init.js

```diff
@@ -107,6 +109,98 @@ export default function init({ servicesManager, configuration }) {
     tools.push(...toolsGroupedByType[toolsGroup])
   );
 
+  /* MeasurementService configuration */
+  const csToolsVer4MeasurementSource = MeasurementService.createSource('CornerstoneTools', '4');
+  const { addOrUpdate, getAnnotation } = csToolsVer4MeasurementSource;
+
+  /* Add mapping criterias to measurement service */
+  const matchingCriteria = {
+    valueType: MeasurementService.VALUE_TYPES.POLYLINE,
+    points: 2,
+  };
+  MeasurementService.addMapping(
+    csToolsVer4MeasurementSource,
+    'Length',
+    matchingCriteria,
+    toAnnotation,
+    toMeasurement
+  );
+
+  /* Measurement Service Events */
+  cornerstone.events.addEventListener(
+    cornerstone.EVENTS.ELEMENT_ENABLED,
+    event => {
+      const {
+        MEASUREMENT_ADDED,
+        MEASUREMENT_UPDATED,
+      } = MeasurementService.EVENTS;
+
+      MeasurementService.subscribe(
+        MEASUREMENT_ADDED,
+        ({ source, measurement }) => {
+          if (![csToolsVer4MeasurementSource.id].includes(source.id)) {
+            const annotation = getAnnotation('Length', measurement.id);
+
+            console.log(
+              '[subscriber::MEASUREMENT_ADDED] Measurement added',
+              measurement,
+            );
+            console.log('Mapped annotation to be saved:', annotation);
+          }
+        });
+
+      MeasurementService.subscribe(
+        MEASUREMENT_UPDATED,
+        ({ source, measurement }) => {
+          if (![csToolsVer4MeasurementSource.id].includes(source.id)) {
+            const annotation = getAnnotation('Length', measurement.id);
+
+            console.log(
+              '[subscriber::MEASUREMENT_UPDATED] Measurement updated',
+              measurement
+            );
+            console.log('Mapped annotation to be saved:', annotation);
+          }
+        }
+      );
+
+      const addOrUpdateMeasurement = eventData => {
```

Comment:
Does our `event` or `eventData` contain the tool's name/key? Note `'Length'` below

Reply:
Updated, thank you